.golangci-kal.yml

```diff
@@ -0,0 +1,55 @@
+run:
+  timeout: 10m
+  go: "1.23"
+  allow-parallel-runners: true
+
+linters:
+  disable-all: true
+  enable:
+    - kal # linter for Kube API conventions
+   
+linters-settings:
+  custom:
+    kal:
+      type: "module"
+      description: KAL is the Kube-API-Linter and lints Kube like APIs based on API conventions and best practices.
+      settings:
+        linters:
+          enable:
+          - "conditions" # Ensure conditions have the correct json tags and markers.
```

Comment:
+1

Seems like the linter also enforces:
```
// +patchStrategy=merge
// +patchMergeKey=type
```
Does this make sense for CRDs?

Obviously we can't fix this finding for the old condition fields that use clusterv1.Conditions

Reply:
I believe those annotations still work for client-side implementations of patching. The patch strategy and key are part of the K8s extension of the OpenAPI v3 schema, so assuming controller-gen is capable of putting these in the right place , they can still be effective for clients like Kubectl patch

Looking at https://tech.aabouzaid.com/2022/11/set-openapi-patch-strategy-for-kubernetes-custom-resources-kustomize.html and other sources of generated CRDs, I don't think controller-gen currently populates these, but it looks like it would work if it did!

Perhaps dropping the patch keys for CRDs is appropriate, I'll check this with api-machinery folks, this could just be a bug in the linter

Reply:
<img width="1089" alt="image" src="https://github.com/user-attachments/assets/06540a9f-2562-4bc8-a950-f8dc3ebbc4b9" />

https://kubernetes.slack.com/archives/C0EG7JC6T/p1718717630176899

Reply:
Then that's a bug in KAL, and I will get that fixed! Thanks for highlighting

Reply:
Thx!

Reply:
Added https://github.com/JoelSpeed/kal/pull/37, which I'll update to once merged.

This will allow us to forbid Proto and PatchStrategy markers from our types, which I think is appropriate as I don't think we will be importing any of our types into core types any time soon